pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/impl/LeastResourceUsageWithWeight.java

```diff
@@ -139,7 +139,8 @@ public Optional<String> selectBroker(Set<String> candidates, BundleData bundleTo
         final double diffThreshold =
                 conf.getLoadBalancerAverageResourceUsageDifferenceThresholdPercentage() / 100.0;
         brokerAvgResourceUsageWithWeight.forEach((broker, avgResUsage) -> {
-            if (avgResUsage + diffThreshold <= avgUsage) {
+            if ((avgResUsage + diffThreshold <= avgUsage)
+                    && candidates.contains(broker)) {
                 bestBrokers.add(broker);
             }
         });
```

Comment:
We can change it to be more simple, right?
```java
for (String broker : candidates) {
    Double avgResUsage = brokerAvgResourceUsageWithWeight.getOrDefault(broker, Double.MAX_VALUE);
    if ((avgResUsage + diffThreshold <= avgUsage)) {
        bestBrokers.add(broker);
    }
}
```

Reply:
you're right, `candidates` is smaller than `brokerAvgResourceUsageWithWeight`, changed it.